Utilities/types.h

```diff
@@ -84,6 +84,27 @@
 #define AUDIT(...) ((void)0)
 #endif
 
+#if defined(__cpp_lib_bit_cast) && (__cpp_lib_bit_cast >= 201806L)
+#include <bit>
+#else
+namespace std
+{
+	template <class To, class From, typename = std::enable_if_t<sizeof(To) == sizeof(From)>>
+	constexpr To bit_cast(const From& from) noexcept
+	{
+		static_assert(sizeof(To) == sizeof(From), "std::bit_cast<>: incompatible type size");
+
+		union aliased
+		{
+			From from;
+			char data[sizeof(From)];
+			To result;
+		};
+
+		return aliased{from}.result;
```

Comment:
From cppreference:
```cpp
template <class To, class From>
typename std::enable_if<
    (sizeof(To) == sizeof(From)) &&
    std::is_trivially_copyable<From>::value &&
    std::is_trivial<To>::value,
    // this implementation requires that To is trivially default constructible
    To>::type
// constexpr support needs compiler magic
bit_cast(const From &src) noexcept
{
    To dst;
    std::memcpy(&dst, &src, sizeof(To));
    return dst;
}
```

Reply:
I doubt an explicit memcpy call would get optimized away by the compiler unlike unions which are likely to be.

Reply:
@elad335 why doubt? Just check. Not only `union` invokes UB here, but it also seems to generate worse code for MSVC (before inlining): https://godbolt.org/z/p2dfEF

Reply:
OK, I'll change it to memcpy.